### DISCUSSION TITLE
Following NEP29

### DISCUSSION THREAD

**Discussion by jules-ch** [pvlib/pvlib-python#1503] [Ideas]
Most scientific packages now follows NEP29 (https://numpy.org/neps/nep-0029-deprecation_policy.html).

This ensures compatibility between dependencies, time based support with numpy & capacity to move forward for packages that are dependent from pandas/scipy/numpy & such.

This will also set a clear deprecation / minimal version dependencies support.

As of now Minimal versions are:

Python 3.8+ & Numpy 1.20+

After looking at PYPI stats, python 3.7 is still the most used version with 80% of downloads.

https://pypistats.org/packages/pvlib

**Comment by kandersolar** [comment]
We do take NEP-29 into consideration (e.g. https://github.com/pvlib/pvlib-python/issues/1491#issuecomment-1184962355), but we don't necessarily follow it 100%, nor do I think we should.  I think @wholmgren's comment here is still the way to go: https://github.com/pvlib/pvlib-python/issues/828#issuecomment-561721447

So long as it's not a maintenance burden on us, what's the downside to continuing to support older versions of dependencies and python itself?

**Reply by wholmgren** [reply]
The maintenance burden of the test framework is currently a good reason to move forward with versions at NEP 29 limits for dependencies. For example, #1491 and #1447

**Comment by jules-ch** [comment]
I completely get the "why drop support when there is no need to". It's more about moving as a whole with the community.
Like supporting a higher more stable pandas version that only supports python 3.7/8+ then pvlib will need to update minimal dependencies.

I support the view of looking at NEP 29 but still taking into consideration usage (see my comment above).

**Comment by wholmgren** [comment]
I came across this nice NEP 29 status page: https://scientific-python.org/specs/spec-0000/
